gateway-test-utils/src/setup.rs

```diff
@@ -104,7 +104,9 @@ pub async fn test_setup_with_config(
     let localhost_port_0 = SocketAddrV6::new(Ipv6Addr::LOCALHOST, 0, 0, 0);
     let mut addresses = vec![localhost_port_0];
     if let Some(addr) = listen_addr {
-        addresses.push(addr);
+        if addr != localhost_port_0 {
```

Comment:
This doesn't seem like a problem but I'm not sure what's going on here.  It seems like the `listen_addr` arg is an extra address on which MGS is supposed to listen, and we always configure it to listen on `[::1]:0`.  Why would the caller pass a `Some` with the same address here?

Reply:
I think this is just me not completely understanding all the test utils we have; looking at it again I'm being overly explicit in the tests and can clean that up (which will make this change unnecessary - I'll revert it).